### ISSUE TITLE
Once upon a time, there was a little girl named Lily

### ISSUE THREAD

**Issue by GilesBathgate** [karpathy/llama2.c#465]
There seems to be a lot of repetition of this in the training data

```bash
grep -c "Once upon a time, there was a little girl named Lily" tinystories.txt 
53467
```

Does anyone know of ways of making qualitative measurements of the training data, i.e diversity of text, (aside from how well it compresses)

**Comment by GilesBathgate** [comment]
Using some simple bash scripts I found:

```
Count, Prefix
53467 Once upon a time, there was a little girl named Lily
30640 Once upon a time, there was a little girl named Sue
20910 Once upon a time, there was a little girl named Mia
19419 Once upon a time, there was a little girl named Lucy
17045 Once upon a time, there was a little girl named Amy
4546 Once upon a time, there was a little girl named Sally
2409 Once upon a time, there was a little girl named Jane
1729 Once upon a time, there was a little girl named Emma
1414 Once upon a time, there was a little girl named Lisa
1257 Once upon a time, there was a little girl named Anna
```

**Comment by GilesBathgate** [comment]
Here is a quick hack to attempt to remove the bias: https://gist.github.com/GilesBathgate/a7a0a18276a2a79836cb6cb44d8656c2

**Comment by GilesBathgate** [comment]
So weird, that ChatGPT has this bias. There are many ways to show it aside from the obvious: https://chat.openai.com/share/f774a1d4-1940-4249-bf1a-642e1ab4ef8f

**Comment by GilesBathgate** [comment]
@karpathy Could this be a suitable starting point for implementing nano RLHF (where the 'Human feedback' is simply some kind of contrastive loss function that penalises repeating prefixes)

**Comment by SenatorAP28** [comment]
Hello...
It could be a suitable start and am new to the tech world
